### ISSUE TITLE
[Doc]: add old sphinx tutorial to our devdocs

### ISSUE THREAD

**Issue by jklymak** [matplotlib/matplotlib#23903]
### Problem

We currently have an interesting tutorial at https://matplotlib.org/sampledoc/index.html.   However, it is really old and wrong in places (https://github.com/matplotlib/sampledoc/issues/26) and we think some of this should be in the main developer docs.   The todo here is to maybe condense and move this sphinx how-to to the main developer documentation.    Probably not too hard, but whoever does it should have some sphinx familiarity and check that it works.

**Comment by timhoffm** [comment]
The only thing I would keep is „Inserting matplotlib figures“. That’s specific to us. Everything else is standard Sphinx and I don’t see a reason why we should bother to maintain that.

**Comment by melissawm** [comment]
Here's a proposal:
- Moving the mpl specific stuff to https://matplotlib.org/stable/devel/documenting_mpl.html
- Linking to https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html (which is the most recent sphinx tutorial that I know of, and pretty nice)

Is this a reasonable way to move forward?

**Comment by melissawm** [comment]
I agree @story645 and after writing this proposal had been thinking of using it as an opportunity to review those docs but I think those are in good shape today. Thanks!
